src/vulkan/engine_vulkan.cc

```diff
@@ -185,12 +189,14 @@ Result EngineVulkan::SetBuffer(BufferType type,
   if (!pipeline_)
     return Result("Vulkan::SetBuffer no Pipeline exists");
 
-  // TODO(jaebaek): Doublecheck those buffers are only for the graphics
-  //                pipeline.
   if (!pipeline_->IsGraphics())
     return Result("Vulkan::SetBuffer for Non-Graphics Pipeline");
 
-  pipeline_->AsGraphics()->SetBuffer(type, location, format, values);
+  if (!vertex_buffer_)
```

Comment:
What happens if it's a kIndex buffer? Why do we create and set a vertex buffer in that case?

Reply:
This is a mistake. Updated.